packages/ag-charts-community/src/util/dom.ts

```diff
@@ -1,4 +1,48 @@
-export function injectStyle(document: Document, cssStyle: string) {
+const verifiedGlobals = {} as { document: Document; window: Window };
+
+if (typeof window !== 'undefined') {
+    verifiedGlobals.window = window;
+} else if (typeof global !== 'undefined') {
+    verifiedGlobals.window = global.window;
+} else {
+    throw new Error('AG Charts - unable to resolve global window.');
+}
+if (typeof document !== 'undefined') {
+    verifiedGlobals.document = document;
+} else if (typeof global !== 'undefined') {
+    verifiedGlobals.document = global.document;
+} else {
+    throw new Error('AG Charts - unable to resolve global document.');
+}
+
+export function getDocument() {
+    return verifiedGlobals.document;
+}
+
+export function getWindow(): Window;
+export function getWindow<K extends keyof Window>(propertyName: K): Window[K];
+export function getWindow<K extends string>(propertyName: K): unknown;
+export function getWindow<K extends keyof Window>(propertyName?: K) {
+    return propertyName ? verifiedGlobals.window[propertyName] : verifiedGlobals.window;
+}
+
+export function createElement<K extends keyof HTMLElementTagNameMap>(tagName: K, options?: ElementCreationOptions) {
+    return getDocument().createElement<K>(tagName, options);
+}
+
+export function downloadUrl(dataUrl: string, fileName: string) {
+    const { body } = getDocument();
+    const element = createElement('a');
+    element.href = dataUrl;
+    element.download = fileName;
+    element.style.display = 'none';
+    body.appendChild(element);
+    element.click();
+    setTimeout(() => body.removeChild(element));
+}
+
+export function injectStyle(cssStyle: string) {
+    const document = getDocument();
```

Comment:
This won't work for server-side rendering (e.g. thumbnail rendering), we need a non-global `Document` in those cases.

Reply:
Discussed offline, we'll run with the changes and see the repurcussions.